src/Module/Module.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace RebelCode\Modular\Module;
+
+/**
+ * Basic implementation of a module.
+ *
+ * @since [*next-version*]
+ */
+class Module extends AbstractCallbackModule implements ModuleInterface
+{
+    /**
+     * Constructor.
+     *
+     * @since [*next-version*]
+     *
+     * @param string   $key          The module key.
+     * @param array    $dependencies The keys of dependency modules. Default: array()
+     * @param array    $config       The configuration. Default: array()
+     * @param callable $onLoad       The callback to invoke when the module is loaded. Default: null
+     */
+    public function __construct(
+        $key,
+        array $dependencies = array(),
+        array $config = array(),
+        callable $onLoad = null
+    ) {
+        $this->_setKey($key)
+            ->_setDependencies($dependencies)
+            ->_setConfig($config)
+            ->_setCallback($this->_normalizeCallback($onLoad));
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    public function getKey()
+    {
+        return $this->_getKey();
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    public function getDependencies()
+    {
+        return $this->_getDependencies();
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    public function getConfig()
```

Comment:
@mecha, you're under arrest for violating [article 9 paragraph 7 of the Dhii standard](https://github.com/Dhii/dhii/blob/formalizing-standards/code-standards/php.md#also). What say you in your defence?